osquery/tables/system/tests/linux/pci_devices_tests.cpp

```diff
@@ -16,102 +16,179 @@ using namespace testing;
-      "\t3152  RV370/M22 [Loops Some X300]\n"
-      "\t3154  RV380/M24 GL [Loops IcyhotGL V3200]\n"
-      "\t3155  RV380 GL [IcyhotMV 2400]\n"
-      "\t3171  RV380 GL [IcyhotMV 2400] (Secondary)\n"
-      "\t3e50  RV380 [Some X600]\n"
-      "\t3e54  RV380 GL [IcyhotGL V3200]\n"
-      "\t3e70  RV380 [Some X600] (Secondary)\n"
-      "\t4136  RS100 [Loops ABC 320M]\n"
-      "ffff  Illegal Vendor ID\n" // Device class information below.
-      "\n\n"
-      "# List of known device classes, subclasses and programming interfaces\n"
-      "\n"
-      "# Syntax:\n"
-      "# C class	class_name\n"
-      "#	subclass	subclass_name  		<-- single tab\n"
-      "#		prog-if  prog-if_name  	<-- two tabs\n"
-      "\n"
-      "C 00  Unclassified device\n"
-      "\t00  Non-VGA unclassified device\n"
-      "\t01  VGA compatible unclassified device\n"
-      "C 01  Mass storage controller\n"
-      "\t00  SCSI storage controller\n"
-      "\t01  IDE interface\n");
-
-  PciDB parsed_db(raw_db);
-
-  std::string got;
-
-  // Happy Path Tests
-  parsed_db.getVendorName("8002", got);
-  EXPECT_EQ("Fake Vendor, Inc.", got);
-
-  parsed_db.getModel("8002", "1313", got);
-  EXPECT_EQ("Foobar [Some R7 Rapidfire]", got);
-
-  parsed_db.getModel("8002", "1314", got);
-  EXPECT_EQ("Wrestler HDMI Audio", got);
-
-  parsed_db.getModel("8002", "4136", got);
-  EXPECT_EQ("RS100 [Loops ABC 320M]", got);
-
-  parsed_db.getSubsystemInfo("8002", "1314", "174b", "1001", got);
-  EXPECT_EQ("90K Diffusion Mini", got);
-
-  parsed_db.getSubsystemInfo("8002", "3150", "103c", "0934", got);
-  EXPECT_EQ("nx8220", got);
-
-  parsed_db.getSubsystemInfo("8002", "1714", "103c", "168b", got);
-  EXPECT_EQ("ProBook 4535s", got);
-
-  // Negative Tests
-  got = "";
-
-  parsed_db.getVendorName("8086", got);
-  EXPECT_EQ("", got);
-
-  parsed_db.getModel("8002", "1388", got);
-  EXPECT_EQ("", got);
-
-  parsed_db.getSubsystemInfo("8002", "1714", "103c", "168c", got);
-  EXPECT_EQ("", got);
-
-  // Things below 'ffff' should not be retrievable.
-  parsed_db.getVendorName("ffff", got);
-  EXPECT_EQ("", got);
-
-  parsed_db.getVendorName("C 00", got);
-  EXPECT_EQ("", got);
-
-  parsed_db.getVendorName("C 01", got);
-  EXPECT_EQ("", got);
-
-  parsed_db.getModel("C 00", "00", got);
-  EXPECT_EQ("", got);
+class PciDevicesTest : public ::testing::Test {
+ protected:
+  PciDevicesTest() {}
+
+  static void SetUpTestCase() {
+    std::istringstream test_db_stream(
+        "# Some Test Comment\n"
+        "8002  Fake Vendor, Inc.\n"
+        "\t1312  Foobar\n"
+        "\t1313  Foobar [Some R7 Rapidfire]\n"
+        "\t131b  Wrestler HDMI Audio\n"
+        "\t\t174b 1001  90K Diffusion Mini\n"
+        "174b  Fake Vendor, LLC.\n"
+        "ffff  Illegal Vendor ID\n" // Device class information below.
+
+    );
+
+    pcidb_ = new PciDB(test_db_stream);
+  }
+
+  static void TearDownTestCase() {
+    delete pcidb_;
+    pcidb_ = nullptr;
+  }
+
+  static PciDB* pcidb_;
```

Comment:
Since `SetUpTestCase` is not exactly the constructor, can you `{nullptr}` here?

Reply:
I don't think I can since you can't inline non-const static member AFAIK.   It is currently initialized on line `47`